lib/docker/executeDockerBuild.ts

```diff
@@ -315,16 +331,67 @@ async function pushEnabled(gi: ProjectAwareGoalInvocation, options: DockerOption
     // tslint:disable-next-line:no-boolean-literal-compare
     if (options.push === true || options.push === false) {
         push = options.push;
-    } else if ((!!options.user && !!options.password) || !!options.config) {
+    } else if (options.registries.some(r => !!(r.username && r.password)) || !!options.config) {
         push = true;
     }
     return projectConfigurationValue("docker.build.push", gi.project, push);
 }
 
 function dockerConfigPath(options: DockerOptions, goalEvent: SdmGoalEvent): string {
-    if (!!options.user && !!options.password) {
+    if (options.registries.some(r => !!(r.username && r.password))) {
         return path.join(os.homedir(), ".docker");
     } else if (!!options.config) {
         return path.join(os.homedir(), `.docker-${goalEvent.goalSetId}`);
     }
 }
+
+function getExternalUrls(tags: string[], options: DockerOptions): ExecuteGoalResult["externalUrls"] {
+    const externalUrls = tags.map(t => {
+        const reg = options.registries.filter(r => t.includes(r.url))[0];
```

Comment:
registries is optional.

Reply:
👍 Will set a default [] for registries as discussed